sql/core/src/test/resources/sql-tests/results/window-with-aggregates.sql.out

```diff
@@ -0,0 +1,526 @@
+  "messageParameters" : {
+    "invalidExprSqls" : "\"count(1)\""
+  },
+  "queryContext" : [ {
+    "objectType" : "",
+    "objectName" : "",
+    "startIndex" : 236,
+    "stopIndex" : 269,
+    "fragment" : "ORDER BY COUNT(*) DESC, col2 + 100"
+  } ]
+}
+
+
+-- !query
+SELECT
+    col1,
+    MAX(col2) as max_col2,
+    LAG(MAX(col2)) OVER (ORDER BY col1) as prev_max,
+    LEAD(MAX(col2)) OVER (ORDER BY col1) as next_max
+FROM VALUES (10, 100), (20, 200), (30, 300), (40, 400) AS t(col1, col2)
+GROUP BY col1
+ORDER BY MAX(col2), col1
+-- !query schema
+struct<>
+-- !query output
+org.apache.spark.sql.catalyst.ExtendedAnalysisException
+{
+  "errorClass" : "UNRESOLVED_COLUMN.WITH_SUGGESTION",
+  "sqlState" : "42703",
+  "messageParameters" : {
+    "objectName" : "`col2`",
+    "proposal" : "`max_col2`, `t`.`col1`, `next_max`, `prev_max`"
+  },
+  "queryContext" : [ {
+    "objectType" : "",
+    "objectName" : "",
+    "startIndex" : 250,
+    "stopIndex" : 253,
+    "fragment" : "col2"
+  } ]
+}
+
+
+-- !query
+SELECT
+    col1 + col2 as sum_expr,
+    COUNT(*) as cnt,
+    PERCENT_RANK() OVER (ORDER BY COUNT(*)) as pct_rank,
+    CUME_DIST() OVER (ORDER BY COUNT(*)) as cum_dist
+FROM VALUES (1, 2), (3, 4), (5, 6), (7, 8) AS t(col1, col2)
+GROUP BY col1 + col2, col1, col2
+ORDER BY COUNT(*), col1 + col2
+-- !query schema
+struct<>
+-- !query output
+org.apache.spark.sql.catalyst.ExtendedAnalysisException
+{
+  "errorClass" : "UNSUPPORTED_EXPR_FOR_OPERATOR",
+  "sqlState" : "42K0E",
+  "messageParameters" : {
+    "invalidExprSqls" : "\"count(1)\""
+  },
+  "queryContext" : [ {
+    "objectType" : "",
+    "objectName" : "",
+    "startIndex" : 261,
+    "stopIndex" : 290,
+    "fragment" : "ORDER BY COUNT(*), col1 + col2"
+  } ]
+}
+
+
+-- !query
+SELECT
+    COALESCE(col1, 0) as safe_col1,
+    COUNT(col2) as non_null_count,
+    SUM(COUNT(col2)) OVER () as total_non_null
+FROM VALUES (1, 10), (NULL, 20), (3, NULL), (4, 40) AS t(col1, col2)
+GROUP BY COALESCE(col1, 0), col1, col2
+ORDER BY COUNT(col2) DESC, COALESCE(col1, 0)
+-- !query schema
+struct<>
+-- !query output
+org.apache.spark.sql.catalyst.ExtendedAnalysisException
+{
+  "errorClass" : "UNSUPPORTED_EXPR_FOR_OPERATOR",
+  "sqlState" : "42K0E",
+  "messageParameters" : {
+    "invalidExprSqls" : "\"count(col2)\""
+  },
+  "queryContext" : [ {
+    "objectType" : "",
+    "objectName" : "",
+    "startIndex" : 234,
+    "stopIndex" : 277,
+    "fragment" : "ORDER BY COUNT(col2) DESC, COALESCE(col1, 0)"
+  } ]
+}
+
+
+-- !query
+SELECT
+    col1,
+    col2,
+    SUM(col1) as sum_col1,
+    AVG(col2) as avg_col2,
+    MAX(SUM(col1)) OVER () as max_sum_col1,
+    MIN(AVG(col2)) OVER () as min_avg_col2
+FROM VALUES (1, 100), (2, 200), (3, 300), (4, 400) AS t(col1, col2)
+GROUP BY col1, col2
+ORDER BY SUM(col1) + AVG(col2), col1, col2
+-- !query schema
+struct<>
+-- !query output
+org.apache.spark.sql.catalyst.ExtendedAnalysisException
+{
+  "errorClass" : "UNSUPPORTED_EXPR_FOR_OPERATOR",
```

Comment:
We have 8 negative test cases for nested aggregate functions as window expressions, is it really necessary?